File: app/src/app/features/recipes/resources/recipe.resource.ts
import { TrackableResource } from 'src/app/core/resources/trackable.resource';
import { RecipeCategoryResource } from './recipe-category.resource';
import { RecipeIngredientResource } from './recipe-ingredient.resource';
import { RecipeStepsResource } from './recipe-step.resource';

export class RecipeResource extends TrackableResource {
  id: number;
  name: string;
  description: string;
  preparationTime: Date;
  fat: number;
  energy: number;
  protien: number;
  carbs: number;
  categories: Array<RecipeCategoryResource>;
  steps: Array<RecipeStepsResource>;
  ingredients: Array<RecipeIngredientResource>;
}

File: api/BeHealth.Business/Resources/FoodManamgnet/Recipes/RecipeResource.cs
using BeHealth.Business.Resources.FoodManamgnet.Recipes;
using BeHealth.Business.Resources.Settings;
using System;
using System.Collections.Generic;

namespace BeHealth.Business.Resources.FoodManamgnet.Meals
{
    public class RecipeResource : TrackableResource
    {
        public long ID { get; set; }
        public string Name { get; set; }

        public string Description { get; set; }

        public DateTimeOffset? PreparationTime { get; set; }

        public double Fat { get; set; }
        public double Energy { get; set; }
        public double Protien { get; set; }
        public double Carbs { get; set; }

        public List<RecipeStepsResource> Steps { get; set; }
        public List<RecipeCategoryResource> Categories { get; set; }
        public List<RecipeIngredientResource> Ingredients { get; set; }

    }
}
